src/modules/module_20400.c

```diff
@@ -0,0 +1,324 @@
+  token.len_max[4] = HASH_LEN_B64;
+  token.attr[4]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_BASE64B;
+
+  const int rc_tokenizer = input_tokenizer ((const u8 *) line_buf, line_len, &token);
+
+  if (rc_tokenizer != PARSER_OK) return (rc_tokenizer);
+
+  // iter
+  const u8 *iter_pos = token.buf[2];
+  salt->salt_iter = hc_strtoul ((const char *) iter_pos, NULL, 10) - 1;
+
+
+  // base64 decode salt
+  const u8 *salt_pos = token.buf[3];
+  const int salt_len = token.len[3];
+  
+  u8 tmp_buf[100] = { 0 };
+  u8 tmp_buf2[100] = { 0 };
+
+  // replace . with + for proper base64 decoding
+  memcpy(tmp_buf2, salt_pos, salt_len);
+  for (int i = 0; i < salt_len; i++)
+  {
+    if (tmp_buf2[i] == '.')
+    {
+      tmp_buf2[i] = '+';
+    }
+  }
+
+  
+  const int salt_len_decoded = base64_decode (base64_to_int, (const u8 *) tmp_buf2, salt_len, tmp_buf);
+
+  u8 *salt_buf_ptr = (u8 *) pbkdf2_sha1->salt_buf;
+  memcpy (salt_buf_ptr, tmp_buf, salt_len_decoded);
+  memcpy (salt->salt_buf, salt_buf_ptr, salt_len_decoded);
+  
+  salt->salt_len    = salt_len_decoded;
+  
+
+  // base64 decode hash
+  const u8 *hash_pos = token.buf[4];
+  const int hash_len = token.len[4];
+
+  // replace . with + for proper base64 decoding
+  memcpy(tmp_buf2, hash_pos, hash_len);
+  for (int i = 0; i < hash_len; i++)
+  {
+    if (tmp_buf2[i] == '.')
+    {
+      tmp_buf2[i] = '+';
+    }
+  }
+
+  base64_decode (base64_to_int, (const u8 *) tmp_buf2, hash_len, tmp_buf);
+  memcpy (digest, tmp_buf, HASH_LEN_RAW);
+
+  digest[0] = byte_swap_32 (digest[0]);
+  digest[1] = byte_swap_32 (digest[1]);
+  digest[2] = byte_swap_32 (digest[2]);
+  digest[3] = byte_swap_32 (digest[3]);
+  digest[4] = byte_swap_32 (digest[4]);
+  
+  return (PARSER_OK);
+}
+
+int module_hash_encode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const void *digest_buf, MAYBE_UNUSED const salt_t *salt, MAYBE_UNUSED const void *esalt_buf, MAYBE_UNUSED const void *hook_salt_buf, MAYBE_UNUSED const hashinfo_t *hash_info, char *line_buf, MAYBE_UNUSED const int line_size)
+{
+  const u32 *digest = (const u32 *) digest_buf;
+
+  const pbkdf2_sha1_t *pbkdf2_sha1 = (const pbkdf2_sha1_t *) esalt_buf;
+
+  // hash
+  u32 tmp[6];
+
+  tmp[0] = byte_swap_32 (digest[0]);
+  tmp[1] = byte_swap_32 (digest[1]);
+  tmp[2] = byte_swap_32 (digest[2]);
+  tmp[3] = byte_swap_32 (digest[3]);
+  tmp[4] = byte_swap_32 (digest[4]);
+  tmp[5] = 0;
+
+  char salt_enc[128] = { 0 };
+  char hash_enc[128] = { 0 };
+
+  int salt_len_enc = base64_encode (int_to_base64, (const u8 *) pbkdf2_sha1->salt_buf, salt->salt_len, (u8 *) salt_enc);
+  int hash_len_enc = base64_encode (int_to_base64, (const u8 *) tmp, HASH_LEN_RAW, (u8 *) hash_enc);
+  
+  
+  // substitute + with . and remove padding =
+  for (int i = 0; i < salt_len_enc; i++) 
+  {
+    if (salt_enc[i] == '+')
+    { 
+      salt_enc[i] = '.';
+    }
+    if (salt_enc[i] == '=') 
+    {
+	  salt_enc[i] = '\0';
```

Comment:
the spaces at the beginning (indentation) are also not very consistent. but I guess this code can be removed after we add/fix the base64_encode () call with the correct base64 chars/table